src/confluent_kafka/src/Admin.c

```diff
@@ -328,93 +328,94 @@ static int Admin_set_replica_assignment (const char *forApi, void *c_obj,
 
         return 1;
 }
+
+
 static int
-Admin_incremental_config_to_c(void *c_obj, PyObject *dict){
+Admin_incremental_config_to_c(PyObject *dict, void *c_obj){
         Py_ssize_t pos = 0;
         PyObject *ko, *vo;
-        while(PyDict_Next(dict,&pos,&ko,&vo)){
-                PyObject *ks, *ks8;
+        PyObject *ks = NULL, *ks8 = NULL;
+        PyObject *vs = NULL, *vs8 = NULL;
+
+        while(PyDict_Next(dict, &pos, &ko, &vo)){
                 const char *k;
-                const char *v = NULL;
-                int op = -1;
+                int config_count = 0;
+                Py_ssize_t i = 0;
                 rd_kafka_error_t *error;
+
                 if (!(ks = cfl_PyObject_Unistr(ko))) {
                         PyErr_Format(PyExc_ValueError,
                                      "expected config name to be unicode "
                                      "string");
-                        return 0;
+                       goto err;
                 }
                 k = cfl_PyUnistr_AsUTF8(ks, &ks8);
 
-                Py_ssize_t iter = 0;
-                PyObject *internal_ko,*internal_vo;
-                PyObject *internal_ks, *internal_ks8;
-                PyObject *internal_vs = NULL, *internal_vs8 = NULL;
-                const char *internal_k;
-                const char *internal_v;
-                while(PyDict_Next(vo,&iter, &internal_ko, &internal_vo)){
-                        if (!(internal_ks = cfl_PyObject_Unistr(internal_ko))) {
-                                PyErr_Format(PyExc_ValueError,
-                                        "expected property name for to be unicode "
-                                        "string");
-                                return 0;
+                if (!PyList_Check(vo) || (config_count = (int)PyList_Size(vo)) < 1) {
+                        PyErr_SetString(PyExc_ValueError,
+                                        "expected non-empty list of entry values");
+                        goto err;
+                }
+
+                for (i = 0; i < config_count; i++) {
+                        PyObject *config_op_value, *config_value,
+                                 *op_type;
+                        Py_ssize_t config_value_cnt;
+                        config_op_value = PyList_GET_ITEM(vo, i);
+                        rd_kafka_AlterConfigOpType_t op;
+                        const char *v = NULL;
+
+                        if (!PyList_Check(config_op_value) ||
+                            (config_value_cnt = (int)PyList_Size(config_op_value)) < 2) {
```

Comment:
`config_value_cnt = (int)PyList_Size(config_op_value)) != 2` instead of `config_value_cnt = (int)PyList_Size(config_op_value)) < 2` as we are expecting exactly 2 values.